### ISSUE TITLE
[Enhancement] Collapse Duplicate Chat

### ISSUE THREAD

**Issue by idle-user** [jackriccomini/spamfilter-plugin-runelite#18]
I originally used the Chat Filter plugin which included an additional feature to collapse messages.  Like so:
![image](https://github.com/jackriccomini/spamfilter-plugin-runelite/assets/15640986/bee3e4ae-f4f7-49d7-8403-fe2f0bb95c73)

The Spam Filter plugin definitely wins the battle with blocking spam messages, as Chat Filter kept having me update the regex lists. I love the feature to just right click and remove it. 

It's a shame the two plug-ins conflict. As having them both on makes the spam messages still pop up in-world.
If Spam Filter added the Collapse Feature, the Chat Filter plug-in will be outshined all together.

**Comment by TreeOtter** [comment]
Hi @idle-user ! I can look at implementing this the next time I'm free if @jackriccomini is OK with the idea. Thanks for the suggestion.

Just to add some additional explanation, the feature this enhancement describes "collapses" duplicate messages. Without this enabled, in the screenshot the Nature Runes message would be displayed twice and the Death Runes three times.

**Comment by idle-user** [comment]
That's correct. There are also toggles to handle each chat tab independently if wanted.
